### DISCUSSION TITLE
Add a grace period after unpausing for taiko, catch and mania

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#17928] [Ideas] [answered]
Unlike standard, after unpausing in taiko, catch or mania, gameplay resumes too quickly and gives a hard time for the player to properly recover after the pause, and often leading to unfair misses.

I suggest handling this issue in 2 ways:

### 1. Resume gameplay after unpausing AND after interacting with the game:
this is how standard handles it, with the "click the orange cursor to resume" thing
![orange](https://user-images.githubusercontent.com/62819481/164699626-24fd6ab0-1314-4462-880a-60053bfdac74.png)
It could be implemented the following way:
- "Press a key to resume" - osu!taiko and osu!mania
- "Move the catcher to resume" - osu!catch

### 2. **Add a 1~3 second delay/countdown right after unpausing:**
Stable handles it this way by having a slight delay after unpausing, which gives players room to prepare and recover.

**Comment by integerrr** [comment]
That sounds like a great way to let people cheese high intensity/stamina intensive sections of map by pausing

**Reply by Theighlin** [reply]
That would somewhat defeat the purpose of pausing the play, you should be able to pause and resume without being punished gameplay-wise for it.

Besides, i reckon there are plans to show the number of pauses a score has on the leaderboard/beatmap page.

**Comment by frenzibyte** [comment, marked as answer]
There is a thread open for it in https://github.com/ppy/osu/issues/10202, with both the game interaction and the countdown proposed already at https://github.com/ppy/osu/issues/10202#issuecomment-696212254. You're welcome to add more there if you see anything missing.
